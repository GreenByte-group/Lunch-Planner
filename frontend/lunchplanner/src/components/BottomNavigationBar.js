import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { BottomNavigation, BottomNavigationAction } from '@material-ui/core';
import PlaceIcon from '@material-ui/icons/Place';
import EventIcon from '@material-ui/icons/LocalDining';
import SocialIcon from '@material-ui/icons/Group';
import {getHistory} from "../utils/HistoryUtils";

const styles = {
    root: {
        boxShadow: '0 -2px 4px 0 rgba(51,51,51,0.1), 0 2px 4px 0 rgba(0,0,0,0.5)',
        position: 'relative',
        bottom: 0,
        width: '100%',
    }
};

class BottomNavigationBar extends React.Component {

    constructor(props) {
        super();

        let value = props.value;

        this.state = {
            value: value,
            search: props.search
        };
    }

    componentWillReceiveProps(newProps) {
        if(newProps.value !== undefined && newProps.value !== null) {
            this.setState({
                value: newProps.value,
            });
        }
        if(newProps.search !== undefined && newProps.search !== null) {
            this.setState({
                search: newProps.search,
            })
        }
    }

    handleChange = (event, value) => {
        switch(value) {
            case 0:
                getHistory().push("/app/location", this.state.search);
                break;
            case 1:
                getHistory().push("/app/event", this.state.search);
                break;
            case 2:
                getHistory().push("/app/team", this.state.search);
                break;
        }
    };

    render() {
        const { classes } = this.props;
        const { value } = this.state;

        return (
                <BottomNavigation
                    className={classes.root}
                    value={value}
                    onChange={this.handleChange}
                    showLabels
                >
                    <BottomNavigationAction label="Places" icon={<PlaceIcon />} />
                    <BottomNavigationAction label="Events" icon={<EventIcon />} />
                    <BottomNavigationAction label="Teams" icon={<SocialIcon />}/>
                </BottomNavigation>
        );
    }
}

BottomNavigationBar.propTypes = {
    classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(BottomNavigationBar);