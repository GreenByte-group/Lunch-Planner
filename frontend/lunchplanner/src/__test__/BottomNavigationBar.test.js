import React from 'react';
import BottomNavigationBar from '../components/BottomNavigationBar';
import renderer from 'react-test-renderer';
import {MemoryRouter} from "react-router-dom";

it ("renders correctly", ()=>{
    const tree = renderer.create(
        <MemoryRouter>
            <BottomNavigationBar/>
        </MemoryRouter>

    ).toJSON();
    expect(tree).toMatchSnapshot();
});