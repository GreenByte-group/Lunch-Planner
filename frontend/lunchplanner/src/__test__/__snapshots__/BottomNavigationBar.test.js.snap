// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiBottomNavigation-root-2 BottomNavigationBar-root-1"
>
  <a
    href="/location"
    onChange={[Function]}
    onClick={[Function]}
    selected={false}
    showLabel={true}
    value={0}
  >
    <button
      className="MuiButtonBase-root-8 MuiBottomNavigationAction-root-3"
      disabled={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiBottomNavigationAction-wrapper-6"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-11"
          color={undefined}
          focusable="false"
          viewBox="0 0 24 24"
        >
          <g>
            <path
              d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
            />
          </g>
        </svg>
        <span
          className="MuiBottomNavigationAction-label-7"
        >
          Places
        </span>
      </span>
      <span
        className="MuiTouchRipple-root-17"
      />
    </button>
  </a>
  <a
    href="/event"
    onChange={[Function]}
    onClick={[Function]}
    selected={true}
    showLabel={true}
    value={1}
  >
    <button
      className="MuiButtonBase-root-8 MuiBottomNavigationAction-root-3"
      disabled={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiBottomNavigationAction-wrapper-6"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-11"
          color={undefined}
          focusable="false"
          viewBox="0 0 24 24"
        >
          <g>
            <path
              d="M8.1 13.34l2.83-2.83L3.91 3.5c-1.56 1.56-1.56 4.09 0 5.66l4.19 4.18zm6.78-1.81c1.53.71 3.68.21 5.27-1.38 1.91-1.91 2.28-4.65.81-6.12-1.46-1.46-4.2-1.1-6.12.81-1.59 1.59-2.09 3.74-1.38 5.27L3.7 19.87l1.41 1.41L12 14.41l6.88 6.88 1.41-1.41L13.41 13l1.47-1.47z"
            />
          </g>
        </svg>
        <span
          className="MuiBottomNavigationAction-label-7"
        >
          Events
        </span>
      </span>
      <span
        className="MuiTouchRipple-root-17"
      />
    </button>
  </a>
  <a
    href="/social"
    onChange={[Function]}
    onClick={[Function]}
    selected={false}
    showLabel={true}
    value={2}
  >
    <button
      className="MuiButtonBase-root-8 MuiBottomNavigationAction-root-3"
      disabled={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiBottomNavigationAction-wrapper-6"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-11"
          color={undefined}
          focusable="false"
          viewBox="0 0 24 24"
        >
          <g>
            <path
              d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
            />
          </g>
        </svg>
        <span
          className="MuiBottomNavigationAction-label-7"
        >
          Social
        </span>
      </span>
      <span
        className="MuiTouchRipple-root-17"
      />
    </button>
  </a>
</div>
`;
